framework/src/multiapps/TransientMultiApp.C

```diff
@@ -210,16 +210,13 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
         bool local_first = _first;
 
         // Now do all of the solves we need
-        while (true)
+        while (!at_steady && ex->getTime() + app_time_offset + 2e-14 < target_time)
```

Comment:
This is probably OK but we might want to make a helper function for this condition at some point?